remoteappmanager/db/orm.py

```diff
@@ -0,0 +1,142 @@
+import contextlib
+
+from sqlalchemy import Column, Integer, Boolean, Unicode, ForeignKey
+from sqlalchemy.exc import OperationalError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, relationship
+from sqlalchemy import create_engine
+
+from remoteappmanager.logging.logging_mixin import LoggingMixin
+
+Base = declarative_base()
+
+
+class UserTeam(Base):
+    """ The user (n <-> n) team association table """
+    __tablename__ = "user_team"
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey('user.id'))
+    team_id = Column(Integer, ForeignKey('team.id'))
+
+
+class User(Base):
+    """ Table for users. """
+    __tablename__ = "user"
+    id = Column(Integer, primary_key=True)
+
+    #: The name of the user as specified by jupyterhub.
+    #: This entry must be unique and is, for all practical purposes,
+    #: a primary key.
+    name = Column(Unicode, index=True, unique=True)
+
+
+class Team(Base):
+    """ Teams of users. """
+    __tablename__ = "team"
+    id = Column(Integer, primary_key=True)
+
+    #: The name of the group. Note that, differently from users, we can have
+    #: multiple teams with the same name. The reason is that we would obtain
+    #: unusual behavior if user A creates a group with name B, and then B
+    #: creates a user. Users are automatically assigned a group with their
+    #: own name when first created.
+    name = Column(Unicode)
+
+    #: The users parts of this team (n <-> n)
+    users = relationship("User", secondary="user_team", backref="teams")
+
+
+class Application(Base):
+    """ Describes an application that should be available for startup """
+    __tablename__ = "application"
+    id = Column(Integer, primary_key=True)
+
+    #: The docker image name where the application can be found
+    image = Column(Unicode)
+
+
+class ApplicationPolicy(Base):
+    __tablename__ = "application_policy"
+    id = Column(Integer, primary_key=True)
+
+    #: If the home directory should be mounted in the container
+    allow_home = Column(Boolean)
+
+    #: If a common workarea should be mounted in the container
+    allow_common = Column(Boolean)
+
+    #: If the container should be accessible from other members of the team
+    allow_team_view = Column(Boolean)
+
+
+class Accounting(Base):
+    """Holds the information about who is allowed to run what."""
+    __tablename__ = "accounting"
+    id = Column(Integer, primary_key=True)
+
+    team_id = Column(Integer, ForeignKey("team.id"))
+
+    application_id = Column(Integer, ForeignKey("application.id"))
+
+    application_policy_id = Column(Integer,
+                                   ForeignKey("application_policy.id"))
+
+    team = relationship("Team")
```

Comment:
Does this work the same as `team = relationship(Team, primaryjoin=team_id==Team.id)`?